### ISSUE TITLE
avoid_redundant_argument_values should ignore required parameters

### ISSUE THREAD

**Issue by passsy** [dart-lang/linter#3034]
```dart 
const ElevatedButton(
  onPressed: null, // causes: avoid_redundant_argument_values
  child: Text('Submit'),
),

// info: Avoid redundant argument values. (avoid_redundant_argument_values at main.dart:27)
```

This warning isn't correct, because `onPressed` is required and can't be avoided.

```dart
class ElevatedButton extends ButtonStyleButton {
  /// Create an ElevatedButton.
  ///
  /// The [autofocus] and [clipBehavior] arguments must not be null.
  const ElevatedButton({
    Key? key,
    required VoidCallback? onPressed, // <-- required
    VoidCallback? onLongPress,
    ValueChanged<bool>? onHover,
    ValueChanged<bool>? onFocusChange,
    ButtonStyle? style,
    FocusNode? focusNode,
    bool autofocus = false,
    Clip clipBehavior = Clip.none,
    required Widget? child,
  }) : super(
```

```
Dart SDK version: 2.15.0-211.0.dev (dev) (Tue Oct 12 10:59:14 2021 -0700) on "macos_x64"
```

**Comment by pq** [comment]
Hey @passsy.  I hope this has resolved itself for you?  It shouldn't reproduce on latest.

![image](https://user-images.githubusercontent.com/67586/144652547-8aaa43e4-9bf0-4085-8e5f-292b0f61b2ee.png)

Please re-open if you're still seeing issues.

Thanks!
